sdk/core/azure-core/src/main/java/com/azure/core/http/rest/IterableResponse.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import reactor.core.publisher.Flux;
+
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+
+/**
+ * This class provides utility to iterate over values.
+ * @param <T> value
+ */
+public class IterableResponse<T> implements Iterable<T> {
+    private final Flux<T> flux;
+
+    /**
+     * Creates instance given {@link Flux}.
+     * @param flux to iterate over
+     */
+    public IterableResponse(Flux<T> flux) {
+        this.flux = flux;
+    }
+
+    /**
+     * Retrieve the {@link Stream} from response.
+     * @return stream of T
+     */
+    public Stream<T> stream() {
+        return flux.toStream();
+    }
```

Comment:
Document what happens if stream() is called multiple times. Does it work? Same with iterator(). Ensure unit tests are written that test if you get a stream, collect it into a list, and then do the same again, are they equivalent? Etc.